packages/diagram/test/diagram_test.dart

```diff
@@ -0,0 +1,173 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+import 'dart:ui' as ui;
+import 'dart:typed_data';
+
+import '../lib/diagram.dart';
+import 'package:test/test.dart';
+import 'package:flutter/material.dart';
+import 'package:image/image.dart' as image;
+
+Widget buildStaticDiagram(BuildContext context) {
+  return new Container(
+    constraints: new BoxConstraints.tight(new Size(100.0, 50.0)),
+    child: new Text('Diagram'),
+  );
+}
+
+File filenameGenerator(Duration timestamp, int index) {
+  return new File('test_name_${timestamp.inMilliseconds}_$index.png');
+}
+
+class TestAnimatedDiagram extends StatefulWidget {
```

Comment:
this one also seems to actually be stateless

Reply:
Fixed.